js/ad_manager_vast.js

```diff
@@ -94,19 +269,43 @@ OO.Ads.manager(function(_, $) {
       'close' ];
 
     /**
-     * Helper function to verify that the xml has a valid vast ad in it and that it is a valid xml.
-     * @param {xml} vastXML Contains the vast ad data to be parsed.
+     * Helper function to verify that XML is valid
+     * @param {xml} vastXML Contains the vast ad data to be parsed
      * @returns {boolean} Returns true if the xml is valid otherwise it returns false.
      */
-    var isValidVastXML = _.bind(function(vastXML) {
+    this.isValidVastXML = _.bind(function(vastXML) {
+      return this.isValidRootTagName(vastXML) && this.isValidVastVersion(vastXML);
+    }, this);
+
+    /**
+     * Helper function to verify XML has valid VAST root tag.
+     * @param {xml} vastXML Contains the vast ad data to be parsed
+     * @returns {boolean} Returns true if the root tag is valid otherwise it returns false.
+     */
+    this.isValidRootTagName = _.bind(function(vastXML) {
       var rootTagName = (vastXML && vastXML.firstChild) ? vastXML.firstChild.tagName || '' : '';
       if (rootTagName.toUpperCase() != "VAST") {
         OO.log("Invalid VAST XML for tag name: " + rootTagName);
+        this.trackError(this.ERROR_CODES.SCHEMA_VALIDATION, this.wrapperParentId);
+        this.errorType = "schemaValidationError";
```

Comment:
instead of hardcoding this, can you associate the this.ERROR_CODES object with both the name and number?
something like:
this.ERROR_CODES = { SCHEMA_VALIDATION: { id: 101, type: "schemaValidationError" }, ...}